### PR TITLE
Link to each task without an <a> tag

### PR DESCRIPTION
**Pull request by karolinkas**
## Description

An <a> tag can not wrap a <tr> component, therefore we are now linking to each task programmatically. 

Contributes to #1846

### CODE REVIEW

File: src/modules/dashboard/components/TaskList/TaskListItem.tsx [hunk 1]
```diff
@@ -76,40 +79,46 @@ const TaskListItem = ({ data, intl: { formatMessage } }: Props) => {
...
   );
 };
 
 TaskListItem.displayName = displayName;
 
-export default injectIntl(TaskListItem);
+const enhance = compose(
+  withRouter,
+  injectIntl,
+) as any;
```
Comment (chmanie) [inline src/modules/dashboard/components/TaskList/TaskListItem.tsx]: I don't know if that _can_ be solved properly right now but it definitely is a code smell. We introduced a lot of `as any`s during the typescript refactor but that doesn't mean we can add more now. If any we should try to remove as many of those as possible. What's the typescript error if you remove it? Let's find a better solution here.

Comment (karolinkas) [reply src/modules/dashboard/components/TaskList/TaskListItem.tsx]: Yeih, found a type, after trying for a while, I will add it in, in other occurrences as well.

**Comment by chmanie** [comment]
There's https://www.npmjs.com/package/@types/recompose, are we using this?

And also this guide (for flow though): https://github.com/acdlite/recompose/tree/master/types

**Comment by ceolson01** [comment]
@chmanie I've narrowed the scope of this PR. I think the type changes to the HOCs should come separately, as the changes are a bit more involved than this PR should see.

This is good for another look.

**Pull request merged**